src/mqt/predictor/reward.py

```diff
@@ -69,3 +70,90 @@ def calc_qubit_index(qargs: list[Qubit], qregs: list[QuantumRegister], index: in
             return qubit_index
     error_msg = f"Global qubit index for local qubit {index} index not found."
     raise ValueError(error_msg)
+
+
+def expected_success_probability(qc: QuantumCircuit, device: Device, precision: int = 10) -> float:
+    """Calculates the expected success probability of a given quantum circuit on a given device.
+
+    Args:
+        qc (QuantumCircuit): The quantum circuit to be compiled.
+        device(mqt.bench.Device): The device to be used for compilation.
+        precision (int, optional): The precision of the returned value. Defaults to 10.
+
+    Returns:
+        float: The expected success probability of the given quantum circuit on the given device.
+    """
+    res = 1.0
+
+    # collect gate durations for all qubits
+    op_times = []
+    for op in device.basis_gates:
+        if op == "barrier" or op == "id":
+            continue
+        if op in ["rxx", "rzz", "cx", "cz", "cp", "ecr", "xx_plus_yy"]:
+            for q0, q1 in device.coupling_map:  # multi-qubit gates
+                try:
+                    d = device.get_two_qubit_gate_duration(op, q0, q1)
```

Comment:
Can we refer to that variable `d` as `duration` to make it easier to understand?